### DISCUSSION TITLE
Build Wiki

### DISCUSSION THREAD

**Discussion by N0cynym** [jglim/CaesarSuite#20] [Ideas]
I think documentation and knowledge are the one and only to let this amazing project create a free high level open source software.

Is it possible to create a wiki to bring in our knowledge and the manual how to use this software?

**Comment by jglim** [comment]
Sounds good, do you have an idea of the content and pages that might be good to include? I drafted this outline, after hearing your opinion and getting a general wiki structure, I will be glad to create the blank wiki pages.

## Welcome (Onboarding)

- Setting up CaesarSuite (requirements, cbf files)
- Reminder about copyrighted files, so that no one accidentally takes down the project
- Differences from Vediamo (for seasoned users to understand the limitations of Diogenes e.g. no flashing, HSCAN only)

## Common Workflows

- Loading a CBF file, and initiating contact
- ECU preparations (e.g. what sort of issues to preempt)

### VC:
- Backup variant strings
- Write new variant strings

### Diagnostic Services:
- If the general diagservice has a valid payload (e.g. not blank or 00 00..), they can be executed normally (e.g. FN_Aktivieren)
- Limitations (no embedded scripts)

### CFF:
- Export CFF segments
- Splice CFF segments

### Memory Editor:
- Read, write memory, adjust parameters

## Troubleshooting

- 7F: ECU unlocked? 
- Saving traces
- How to ask for help

## Technical Notes

- Looking into c32s.dll
- Communications (brief on J2534)
- Protocol support (UDS, KW2C3PE)
- Fingerprinting
- Diogenes behavior for VC (SCN zeroing)
- Preparing a bench setup

**Reply by N0cynym** [reply]
Yes that's definitely what I'm meaning.

**The first page should let us explain CaesarSuite and point on the copyrigths!**

![CaesarSuiteWiki](https://user-images.githubusercontent.com/75754060/104103443-6d7cb100-52a2-11eb-899a-2f6e717858ce.png)



[The Editor](https://app.diagrams.net/) 
[CaesarSuiteWiki.zip](https://github.com/jglim/CaesarSuite/files/5791423/CaesarSuiteWiki.zip)

**Comment by jglim** [comment]
Got it. I'll try to figure out how to enable the wiki and setup the pages within these few days. Thanks for looking into it and coming up with the nice diagrams

**Reply by N0cynym** [reply]
You're welcome :wink:
I updated [this](https://github.com/jglim/CaesarSuite/discussions/20#discussioncomment-268766) a little bit.

**Reply by jglim** [reply]
I looked at the tree for a bit, it seems like there is some overlap between User Manual and Guides, so I tried to aggregate them together. 

The wiki is now up with significant change to the layout, with notes to refer back to your original diagram (simpler is preferred for now!); hope it looks alright. Your account should have edit capabilities, let me know again if it doesn't work and I will check the repository's settings.

**Reply by N0cynym** [reply]
Looks like I'm not able to edit.
![grafik](https://user-images.githubusercontent.com/75754060/104149776-d285f300-53d7-11eb-9248-a71d7b2a425e.png)

**Reply by jglim** [reply]
I changed the settings again, please let me know if it is still broken. Not sure how Github's wiki access control works, so I changed it to be extra permissive for now.

**Reply by N0cynym** [reply]
Thanks :smile:  
I will give my best to upload good content as soon as possible.
![grafik](https://user-images.githubusercontent.com/75754060/104153952-eafc0a80-53e3-11eb-8656-dd3254f3c7d9.png)

**Comment by N0cynym** [comment]
Does someone know how to upload images to wiki?
Is the only way to store them in a private repository or where will/must we store them?

**Reply by N0cynym** [reply]
This is my latest edit from [draw.io](draw.io)
[BasicWorkflowCaesarSuiteObd.zip](https://github.com/jglim/CaesarSuite/files/5817102/BasicWorkflowCaesarSuiteObd.zip)

![BasicWorkflowCaesarSuiteObdJ2534](https://user-images.githubusercontent.com/75754060/104651392-dc4d7600-56b7-11eb-8543-e791d2fc675e.png)

**Reply by jglim** [reply]
This issue has been created for anyone to post images inside, and can be used together with the wiki : https://github.com/jglim/CaesarSuite/issues/23 ; images can be uploaded there, and referenced in the wiki. Hope it makes the process somewhat easier.

Quick note about your J2534 question, the standard is "officially" available from the standards body, SAE, but is behind a paywall. I found [these documentation from DrewTech](http://www.drewtech.com/support/passthru.html) to be a very good alternative.

**Reply by N0cynym** [reply]
Thank you. [KVASER](https://www.kvaser.com/about-can/can-standards/j2534/) have also a good english description about J2534.
I will search for more and update the wiki as soon as I'm able for.
